miner/block.go

```diff
@@ -128,7 +128,19 @@ func prepareBlock(w *worker) (*blockState, error) {
 		if (lastCommitment != common.Hash{}) {
 			lastRandomnessParentHash := rawdb.ReadRandomCommitmentCache(w.db, lastCommitment)
 			if (lastRandomnessParentHash == common.Hash{}) {
-				return b, errors.New("Failed to get last randomness cache entry")
+				log.Warn("Randomness cache miss while building a block. Attempting to recover.", "number", header.Number.Uint64())
+
+				// We missed on the cache which should have been populated, attempt to repopulate the cache.
+				err := w.chain.RecoverRandomnessCache(lastCommitment, b.header.ParentHash)
+				if err != nil {
+					log.Error("Error in recovering randomness cache", "error", err, "number", header.Number.Uint64())
```

Comment:
mind that error messages are to be lowercase by golang convention. we are probably not following too much anyway

Reply:
Made an edit to that 9e21aa8cc0a6e2a54dbc039da44a85ed1b4efa97